src/components/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+
+type Props = {
+  query: string,
+  handleQueryChange: (event: React.ChangeEvent<HTMLInputElement>) => void,
+  setIsDropdownVisible: (value: boolean) => void,
+};
+
+export const SearchBar: React.FC<Props> = ({
+  query,
+  handleQueryChange,
+  setIsDropdownVisible,
+}) => (
+  <div className="dropdown-trigger">
+    <input
+      onFocus={() => setIsDropdownVisible(true)}
```

Comment:
To make it perfect - it is best to handle `onBlur` - to hide the suggestion, when component looses the focus

Reply:
I have added it, however to make it work we need to add some delay before the dropdown collapses. It is needed to have the opportunity to handle the button click.

Reply:
Looks like a hack :)

Ok, though I believe there are other ways to fix this, IMO your solution is not bad, so let's keep it 👍